tests/selenium/test_basic_copy_paste.py

```diff
@@ -0,0 +1,212 @@
+import dash
+from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
+
+import dash_core_components as dcc
+import dash_html_components as html
+from dash_table import DataTable
+
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.action_chains import ActionChains
+
+import pandas as pd
+
+url = "https://github.com/plotly/datasets/raw/master/" "26k-consumer-complaints.csv"
+rawDf = pd.read_csv(url)
+df = rawDf.to_dict('rows')
+
+def get_app():
+    app = dash.Dash(__name__)
+
+    app.layout = html.Div(
+        [
+            DataTable(
+                id="table",
+                data=df[0:250],
+                columns=[{"name": i, "id": i, "hideable": i == "Complaint ID"} for i in rawDf.columns],
+                editable=True,
+                sort_action='native',
+                include_headers_on_copy_paste=True,
+            ),
+            DataTable(
+                id="table2",
+                data=df[0:10],
+                columns=[{"name": i, "id": i, "deletable": True} for i in rawDf.columns],
+                editable=True,
+                sort_action='native',
+                include_headers_on_copy_paste=True,
+            ),
+        ]
+    )
+
+    @app.callback(
+        Output("table", "data"),
+        [Input("table", "data_timestamp")],
+        [State("table", "data"), State("table", "data_previous")],
+    )
+    # pylint: disable=unused-argument
+    def update_data(timestamp, current, previous):
+        # pylint: enable=unused-argument
+        if timestamp is None or current is None or previous is None:
+            raise PreventUpdate
+
+        modified = False
+        if len(current) == len(previous):
+            for (i, datum) in enumerate(current):
+                previous_datum = previous[i]
+
+                if datum['Unnamed: 0'] != previous_datum['Unnamed: 0']:
+                    datum['Complaint ID'] = 'MODIFIED'
+                    modified = True
+
+        if modified:
+            return current
+        else:
+            raise PreventUpdate
+
+    return app
+
+
+def test_tbcp001_copy_paste_callback(test):
+    test.start_server(get_app())
+
+    with test.table('table') as target:
+        target.cell.click(0, 0)
```

Comment:
This usage is a bit funny grammatically, makes it look like you have on cell `target.cell` and you're clicking it at `(0, 0)` whatever that means. I suppose it would be a little clearer to say `target.cells.click(0, 0)` but what if we refactor it to something like `target.cell(0, 0).click()`? Same with `.row` and `.column`. But `.paging` is clear as is.

Reply:
I like it

Reply:
https://github.com/plotly/dash-table/pull/696/commits/73092c7b929f4f2b2a3f997dd4bd47e4cc43f699